### PR TITLE
ref(ts) Convert KeyValueList to typescript

### PR DESCRIPTION
**Pull request by markstory**
Convert the un-annotated KeyValueList component to typescript

### CODE REVIEW

File: src/sentry/static/sentry/app/components/events/interfaces/keyValueList/keyValueList.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  isContextData = false,
+  isSorted = true,
+  raw = false,
+  longKeys = false,
+}: Props) {
+  // TODO(dcramer): use non-string keys as reserved words ("unauthorized")
+  // break rendering
+
+  if (data === undefined || data === null) {
+    data = [];
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/events/interfaces/keyValueList/keyValueList.tsx:50]: 👎 to masking parameters, though looks like it already did this, so neautral.

File: src/sentry/static/sentry/app/components/events/interfaces/keyValueList/keyValueList.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  } else if (!(data instanceof Array)) {
+    data = Object.entries(data);
+  } else {
+    data = data.filter(kv => kv !== null);
+  }
+
+  data = isSorted ? sortBy(data, [([key]) => key]) : data;
+  const props = onClick ? {onClick} : {};
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/events/interfaces/keyValueList/keyValueList.tsx:58]: I think we can spread in undefined onClick and it's fine

**Review by evanpurkhiser — APPROVED** [review]
Looks nice and straight forward.

File: src/sentry/static/sentry/app/components/events/interfaces/keyValueList/keyValueList.tsx (hunk 1 above)
Comment (markstory) [reply src/sentry/static/sentry/app/components/events/interfaces/keyValueList/keyValueList.tsx:50]: I agree, I'm not excited by it either. I was aiming to change the runtime code as little as possible. This is part of issue details, handles way too many types of data, and I didn't want to chance a regression

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/events/interfaces/keyValueList/keyValueList.tsx:50]: Yeah understandable

File: src/sentry/static/sentry/app/components/events/interfaces/keyValueList/keyValueList.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+import ContextData from 'app/components/contextData';
+import theme from 'app/utils/theme';
+
+type Props = {
+  /**
+   * Data to render, could be almost anything as the various event interfaces
+   * support all sorts of data.
+   */
+  data: undefined | null | Record<string, any> | any[];
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/events/interfaces/keyValueList/keyValueList.tsx:13]:
```suggestion
  data?: null | Record<string, any> | any[];
```

**Review by priscilawebdev — APPROVED** [review]
looks good to me

**Pull request merged**